### ISSUE TITLE
Can't serialize on iOS

### ISSUE THREAD

**Issue by Garethdl** [jacobdufault/fullserializer#19]
There is probably a way around this, but I can't see it in your documentation.

I get the following error when using with iOS:

ExecutionEngineException: Attempting to JIT compile method '(wrapper delegate-invoke) System.Reflection.MonoProperty/Getter`2<Job, Address>:invoke_Address__this___Job (Job)' while running with --aot-only.

This is the same error I got with the XML serialization (the reason I tried using this one).

**Comment by jacobdufault** [comment]
Can you post some code that causes the exception? Is this possibly related to issue #17 (which is waiting on a Unity fix)?

**Comment by Garethdl** [comment]
Thanks for the reply. This code illustrates the problem... I just made a test class and function to call my Serialize function (which I mostly copied yours).

class testClass
{
	
	public testClass()
	{
		//Empty constructor for serialization
	}

	private int _a;
	[SerializeField]
	public int A
	{
		get { return this._a; }
		set { this._a = value; }
	}
	
}

public static string TestPropertySerialization()
{
	testClass test = new testClass();
	test.A = 5;
	
	string str = SerialiseDataJSON(typeof(testClass), test);
	return str;	
}

public static string SerialiseDataJSON(Type type, object toSerialise)
{
	FullSerializer.fsSerializer _serializer = new FullSerializer.fsSerializer();
	FullSerializer.fsData data;
	
	string msg =  _serializer.TrySerialize(type, toSerialise, out data).FormattedMessages;
	
	
	return FullSerializer.fsJsonPrinter.CompressedJson(data);		
}

If I had of just made the property a public variable, it would have worked.

I get this error message through XCode when using on an iPad Air 2

ExecutionEngineException: Attempting to JIT compile method 'System.Reflection.MonoProperty:GetterAdapterFrame<Shared/testClass, int> (System.Reflection.MonoProperty/Getter`2<Shared/testClass, int>,object)' while running with --aot-only.



  at System.Reflection.MonoProperty.GetValue (System.Object obj, System.Object[] index) [0x00000] in <filename unknown>:0 

  at FullSerializer.Internal.fsMetaProperty.Read (System.Object context) [0x00000] in <filename unknown>:0 

  at FullSerializer.Internal.fsReflectedConverter.TrySerialize (System.Object instance, FullSerializer.fsData& serialized, System.Type storageType) [0x00000] in <filename unknown>:0 

  at FullSerializer.fsSerializer.InternalSerialize_4_Converter (System.Object instance, FullSerializer.fsData& data) [0x00000] in <filename unknown>:0 

  at FullSerializer.fsSerializer.InternalSerialize_3_ProcessVersioning (System.Object instance, FullSerializer.fsData& data) [0x00000] in <filename unknown>:0 

  at FullSerializer.fsSerializer.InternalSerialize_2_Inheritance (System.Type storageType, System.Object instance, FullSerializer.fsData& data) [0x00000] in <filename unknown>:0 

  at FullSerializer.fsSerializer.InternalSerialize_1_ProcessCycles (System.Type storageType, System.Object instance, FullSerializer.fsData& data) [0x00000] in <filename unknown>:0 

  at FullSerializer.fsSerializer.TrySerialize (System.Type storageType, System.Object instance, FullSerializer.fsData& data) [0x00000] in <filename unknown>:0 

  at Shared.SerialiseDataJSON (System.Type type, System.Object toSerialise) [0x00000] in <filename unknown>:0 

  at Shared.TestPropertySerialization () [0x00000] in <filename unknown>:0

**Comment by jacobdufault** [comment]
Thanks! This is an issue with il2cpp (duplicate of #17), please see the Unity bug report [here](http://fogbugz.unity3d.com/default.asp?673824_7trtmak8758gc1h8).

(Why il2cpp? Look at the stack trace -

```
ExecutionEngineException: Attempting to JIT compile method 'System.Reflection.MonoProperty:GetterAdapterFrame (System.Reflection.MonoProperty/Getter`2,object)' while running with --aot-only.

at System.Reflection.MonoProperty.GetValue (System.Object obj, System.Object[] index) [0x00000] in :0
```

the code inside of GetValue is messing up)

**Comment by jacobdufault** [comment]
For now you can just use fields instead of properties as a workaround. Not the greatest, but it'll work for the moment.
